src/sources/util/http/prelude.rs

```diff
@@ -184,15 +191,26 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
 
             let span = Span::current();
             let make_svc = make_service_fn(move |conn: &MaybeTlsIncomingStream<TcpStream>| {
+                let remote_addr = conn.peer_addr();
+                let remote_addr_ref = enable_source_ip.then(|| Arc::new(remote_addr));
```

Comment:
Do you still think the `Arc` is helpful? I know you mentioned before that you didn't think it made a difference. We could drop it if so.

Reply:
Ya, we can explore this idea.

Reply:
yes, seems like removing `Arc` definitely be helpful.

<img width="699" alt="image" src="https://github.com/vectordotdev/vector/assets/51093026/1a961f3b-b4bb-48ae-b33a-67d74843d411">

This was the difference between using `Copy` and `Arc::clone` which I observed